Use RandomNumberGenerator.Fill()

martincostello opened the pull request:
Replace local usages of `RandomNumberGenerator.Create()` in a using block calling `GetBytes()` with a call to the static `RandomNumberGenerator.Fill()` method.

Tornhoof commented on src/libraries/Common/src/System/Marvin.cs:
```diff
@@ -108,12 +108,16 @@ private static uint _rotl(uint value, int shift)
+#else
             using (RandomNumberGenerator rng = RandomNumberGenerator.Create())
             {
-                var bytes = new byte[sizeof(ulong)];
```
does it make sense or is it allowed to use stackalloc here?

martincostello replied:
I don't actually know. It's not something I've used myself.

dannyd89 replied:
This would imply using unsafe which the devs try to avoid if possible. But the files have some small allocations which could be changed to stackalloc tbh

stephentoub replied:
> This would imply using unsafe 

It doesn't. The Fill method takes a span, and stackalloc'ing into a span doesn't require unsafe. We should stackalloc here instead of allocating.

martincostello replied:
So I guess the only relevant change actually in this PR is the changes in this file, plus changing it to a `stackalloc`?

stephentoub commented on src/libraries/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/SymmetricKeyWrap.cs:
```diff
@@ -30,10 +30,14 @@ internal static byte[] TripleDESKeyWrapEncrypt(byte[] rgbKey, byte[] rgbWrappedK
 
             // generate a random IV
             byte[] rgbIV = new byte[8];
+#if NETCOREAPP || NETSTANDARD2_1
```
This library doesn't build for either of these configurations. There's no benefit to adding this.

stephentoub commented on src/libraries/System.Security.Cryptography.Cng/tests/SymmetricCngTestHelpers.cs:
```diff
@@ -288,10 +288,14 @@ internal static bool SupportsPersistedSymmetricKeys
             byte[] buffer = new byte[count];
+#if NETCOREAPP || NETSTANDARD2_1
+            RandomNumberGenerator.Fill(buffer);
+#else
```
This is a test. We should prefer simplicity over hyper optimization. This isn't needed.

stephentoub commented on src/libraries/System.DirectoryServices/src/System/DirectoryServices/ActiveDirectory/TrustHelper.cs:
```diff
@@ -961,9 +961,13 @@ internal static string CreateTrustPassword()
             byte[] buf = new byte[PASSWORD_LENGTH];
             char[] cBuf = new char[PASSWORD_LENGTH];
 
+#if NETCOREAPP || NETSTANDARD2_1
```
This library doesn't build for either of these configurations. There's no benefit to adding this.

martincostello commented:
@stephentoub Thanks for looking - it started as something I though relevant (I recently did the same changes in ASP.NET Core) as I assumed (incorrectly) that only the latest TFMs where in the master branch for 5.0, but I guess these are all actually redundant. Shall i just close this PR?

martincostello committed:
Use RandomNumberGenerator.Fill()

Replace usages of RandomNumberGenerator.Create() with a call to the static RandomNumberGenerator.Fill() method for supported TFMs.
Use stackalloc for the byte array.

martincostello commented:
I've rebased the PR down to just the change to use `Fill()` in `Marvin.cs`, as well as changing to `stackalloc`.

martincostello commented:
I'm not really sure what I'm doing here with the stackalloc, and as it only affects one call once, I'll just close this PR and leave it as-is.
